src/dist/worker.cc

```diff
@@ -236,44 +345,93 @@ agd::Status Worker::Run(const Params& params, const Parameters& aligner_params,
-      // get timestamp, queue size
-      //cout << "queue measure thread starting ...\n";
-      while(run_) {
-        time_t result = std::time(nullptr);
-        timestamps_.push_back(static_cast<long int>(result));
-        queue_sizes_.push_back(work_queue_->size());
-        std::this_thread::sleep_for(std::chrono::milliseconds(500));
-        if (queue_sizes_.size() >= 1000000) {
-          break;  // dont run forever ... 
-        }
+      if (!incomplete_request_queue_->pop(request)) {
+        continue;
+      }
+
+      auto size = request.buf.size();
+      zmq::message_t msg(request.buf.release_raw(), size, free_func, NULL);
+      bool success = zmq_incomplete_request_socket_->send(std::move(msg));
+      if (!success) {
+        cout << "INCOMP REQ Thread failed to send response over zmq!\n";
       }
-      //cout << "queue measure thread finished\n";
     }
-  );
+
+    cout << "Incomplete request queue thread ending.\n";
+  });
+
+  timestamps_.reserve(100000);
+  queue_sizes_.reserve(100000);
+
+  // queue_measure_thread_ = std::thread([this](){
+  //     // get timestamp, queue size
+  //     //cout << "queue measure thread starting ...\n";
+  //     while(run_) {
+  //       time_t result = std::time(nullptr);
+  //       timestamps_.push_back(static_cast<long int>(result));
+  //       queue_sizes_.push_back(work_queue_->size());
+  //       std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  //       if (queue_sizes_.size() >= 1000000) {
+  //         break;  // dont run forever ...
+  //       }
+  //     }
+  //     cout << "queue measure thread finished\n";
+  //   }
+  // );
+
+  // Raises SIGUSR1 in 15 secs time
+  // std::thread partial_measure_thread_ = std::thread([this, &nPartial]()  {
+  //   std::this_thread::sleep_for(std::chrono::milliseconds(15*1000));
+  //   cout << "Number of partial merges: " << nPartial << std::endl;
+  //   //cout << "Raising SIGUSR1 to true\n";
+  //   //raise(SIGUSR1);
+  //   cout << "Partial measure thread ending.\n";
+  // });
+
+  // std::thread validation_thread_ = std :: thread([this, &signal_num](){
+  //   while(!(*signal_num)) {
+  //     std::this_thread::sleep_for(std::chrono::milliseconds(5*1000));
+  //     cout << "Size of work queue: " << work_queue_->size() << std::endl;
+  //     cout << "Size of incomplete response queue: " <<
+  //     incomplete_request_queue_->size() << std::endl; cout << "Size of result
+  //     queue: " << result_queue_->size() << std::endl;
+  //   }
+  // });
+
+  while (!(*signal_num)) {
+    std::this_thread::sleep_for(std::chrono::milliseconds(10 * 100));
+  }
+  cout << "Response [signal_num] value change received.\n";
+  SignalHandler(*signal_num);
 
   cout << "Worker running, press button to exit\n";
-  //std::cin.get();
+  // std::cin.get();
 
   cout << "joining threads ...\n";
-  //run_ = false;
-  //work_queue_->unblock();
-  //result_queue_->unblock();
+  // run_ = false;
```

Comment:
if this stuff isnt needed any more delete it. Should we also add a Ctrl-C handler that works in the same way?

Reply:
Fine!